flow_control/flow.py

```diff
@@ -306,17 +212,56 @@ def studio_view(self, context=None):
         fragment = super(FlowCheckPointXblock,
                          self).studio_view(context=context)
 
-        fragment.add_javascript(load("static/js/injection.js"))  # We could also move this function to a different file
+        # We could also move this function to a different file
+        fragment.add_javascript(self.resource_string("static/js/injection.js"))
         fragment.initialize_js('EditFlowControl')
 
         return fragment
 
-    @XBlock.json_handler
-    def control_point(self, data, suffix=''):
-        """        """
-
-        return {
-            'success': True,
-            'data': data,
-            'suffix': suffix,
-        }
+    def compare_scores(self, correct, total):
+        if total:
+            # getting percentage score for that section
+            percentage = (correct / total) * 100
+
+            if self.operator == 'equal':
+                return percentage == self.ref_value
+            if self.operator == 'distinct':
+                return percentage != self.ref_value
+            if self.operator == 'less than or equal to':
+                return percentage <= self.ref_value
+            if self.operator == 'greater than or equal to':
+                return percentage >= self.ref_value
+            if self.operator == 'less than':
+                return percentage < self.ref_value
+            if self.operator == 'greater than':
+                return percentage > self.ref_value
+
+        return False
+
+    def condition_problem(self, location_str):
+
+        usage_key = UsageKey.from_string(location_str)
+        user_id = self.xmodule_runtime.user_id
+
+        scores_client = ScoresClient(self.course_id, user_id)
+        scores_client.fetch_scores([usage_key])
+        score = scores_client.get(usage_key)
+
+        return self.compare_scores(score.correct, score.total)
+
+    def condition_subsection(self, problems):
```

Comment:
if this is the condition for a list of problems, we better rename it to condition list_of_problems or something more descriptive